### PR TITLE
Fixes #89 Added support for setting the chdir attr

### LINKED ISSUE #89: [FR]: Add support for chdir attribute
### What is the current behavior?

The rule jest_test accepts a chdir attribute as does js_test, however, the implementation of jest_test does not adapt the file paths or arguments created by it to be relative to the chdir argument.

For example, the path to the jest config is auto-generated but not relative to the chdir attribute, and therefore results in an error:

```
Error: Can't find a root directory while resolving a config file path.
Provided path to resolve: path/from/root/to/jest.config.js
```

### Describe the feature

Add support for the chdir attribute to update the generated file paths relative to desired working dir

### Fund our work

- [ ] Sponsor our open source work by donating a [feature bounty](https://opencollective.com/aspect-build/)

### PR DESCRIPTION
**Pull request by twheys**
Added logic to invert the chdir argument and adapt paths passed as cli options to the jest command

### CODE REVIEW

File: jest/private/jest_test.bzl [hunk 1]
```diff
@@ -89,7 +115,7 @@ def _impl(ctx):
         # https://jestjs.io/docs/cli#--configpath. The path to a Jest config file specifying how to
         # find and execute tests.
         "--config",
-        generated_config.short_path,
+        paths.join(inv_chdir, generated_config.short_path),
```
Comment (jbedard) [inline jest/private/jest_test.bzl:118]: I think we can just go with the simple case for now and assume there will be no `.` or `..` in the path. If we decide to support more complex paths we should do it across all rulesets.

See https://docs.aspect.build/rules/aspect_rules_js/docs/js_binary/#chdir for the `..` snippet, or webpack has [a simple one](https://github.com/aspect-build/rules_webpack/blob/04b6a2714bfff1b55dc166048b32e59afd0e9268/webpack/private/webpack_devserver.bzl#L45-L47) you can copy.

Comment (twheys) [reply jest/private/jest_test.bzl:118]: Thanks for the feedback, I updated the commit to reflect how this is done in webpack

**Review by jbedard — APPROVED** [review]

File: jest/private/jest_test.bzl (hunk 1 above)
Comment (jbedard) [reply jest/private/jest_test.bzl]: I think you also need to remove the test containing the `.` and `..`s now though, see the test errors.

File: jest/private/jest_test.bzl [hunk 2]
```diff
@@ -77,6 +78,11 @@ def _impl(ctx):
         },
     )
 
+    # Unwind the chdir argument to adapt the path arguments
+    unwind_chdir_prefix = ""
+    if ctx.attr.chdir:
+        unwind_chdir_prefix = "/".join([".."] * len(ctx.attr.chdir.split("/"))) + "/"
```
Comment (gregmagolan) [inline jest/private/jest_test.bzl:84]: We've written this `../../..` generating code so many times; lets upstream it to a bazel-lib utility function one day :)

Comment (jbedard) [reply jest/private/jest_test.bzl:84]: Yeah that's what I was thinking, and when we do we can think about maybe handling `.`, `..` etc

**Pull request merged**